src/exchangeFsm.js

```diff
@@ -161,12 +161,12 @@ const Factory = function (options, connection, topology, serializers, exchangeFn
         );
         return Promise.reject(new Error('Exchange has reached the limit of messages waiting on a connection'));
       }
-      var publishTimeout = message.timeout || options.publishTimeout || message.connectionPublishTimeout || 0;
+      const publishTimeout = message.timeout || options.publishTimeout || message.connectionPublishTimeout || 0;
       return new Promise(function (resolve, reject) {
-        var timeout;
-        var timedOut;
-        var failedSub;
-        var closedSub;
+        let timeout;
+        let timedOut;
+        let failedSub = null;
+        let closedSub = null;
```

Comment:
and here

Reply:
why set to `null`? Was this a linting suggestion? Seems fine but just curious